syntax = "proto3";

package replication;

service ReplicationService{
    rpc GetTopic (GetTopicRequest) returns (GetTopicResponse);
    rpc GetQueue (GetQueueRequest) returns (GetQueueResponse);
    rpc FailOver (FailOverRequest) returns (FailOverResponse);
}

message GetTopicRequest{
    string topic_id = 1;
}

message GetTopicResponse{
    bool success = 1;
    string message = 2;
    Topic topic = 3;
}

message GetQueueRequest{
    string queue_id = 1;
}

message GetQueueResponse{
    bool success = 1;
    string message = 2;
    Queue queue = 3;
}

message FailOverRequest{
    string target = 1;
    bool isTopic = 2;
    string id = 3;
}

message FailOverResponse{
    bool success = 1;
    string message = 2;
}

message Topic {
    string topic_id = 1;
    string autor = 2;
    repeated Message messages = 3;
}

message Queue {
    string queue_id = 1;
    string autor = 2;
    repeated Message messages = 3;
}

message Message {
    string message_id = 1;
    string parent = 2;
    string content = 3;
    string sender = 4;
    string timestamp = 5;
}
